build-system/server/app.js

```diff
@@ -45,6 +46,9 @@ const {replaceUrls, isRtvMode} = require('./app-utils');
 const TEST_SERVER_PORT = argv.port || 8000;
 let SERVE_MODE = getServeMode();
 
+// Envelopes require rewriting these DNS records
+evilDns.add('*.localhost', '127.0.0.1');
```

Comment:
Is this applicable under all circumstances where a server is launched? (`gulp`, `gulp serve`, `gulp e2e`, `gulp integration` all use this)

Reply:
Yes, but I can't think of a case where it hurts.

Reply:
As a future task, we could consider replacing the entire `hosts` section of `.travis.yml` with this.

Reply:
As a future task, we could consider replacing the entire `hosts` section of `.travis.yml` with this.

Reply:
@rsimha I believe the other entries on .travis.yml are used by browsers directly (e.g: fonts.googleapi.com), so they require a different solution. Puppeteer supports HTTP request interception but Selenium does not. The only thing I think that would work today is to start a proxy server and direct all browsers to use it, and at the proxy we could do various things to limit Internet access. I can try it and see how well it works.